include/service/entities/custom_object_service.js

```diff
@@ -955,7 +955,7 @@ module.exports = function CustomObjectServiceModule(pb) {
             options = {};
         }
 
-        if (!pb.validation.isIdStr(id, true)) {
+        if (!pb.validation.isIdStr(id.toString(), true)) {
```

Comment:
Then it won't accept strings. I think we need to have both checks here.
On Nov 16, 2015 5:34 PM, "Brian" <notifications@github.com> wrote:

> In include/service/entities/custom_object_service.js
> <https://github.com/pencilblue/pencilblue/pull/861#discussion_r44994990>:
>
> > @@ -955,7 +955,7 @@ module.exports = function CustomObjectServiceModule(pb) {
> >              options = {};
> >          }
> >
> > -        if (!pb.validation.isIdStr(id, true)) {
> > +        if (!pb.validation.isIdStr(id.toString(), true)) {
>
> You should be able to change this to if (!pb.validation.isId(id, true)) {.
> Otherwise you risk a null reference error
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pencilblue/pencilblue/pull/861/files#r44994990>.
>